From b69bfb2a30df65e7f267dd6998276fa3986bd14b Mon Sep 17 00:00:00 2001
From: Ralph Giles <giles@thaumas.net>
Date: Sun, 14 Jun 2020 00:53:30 -0700
Subject: [PATCH] Fix arm build with rtcd enabled.

The autotools build doesn't set OPUS_HAVE_RTCD for arm targets,
assuming all the supported intrinsics will work on the runtime
cpu.

The cmake build however defines this by default when the neon
extension is available on the target. On Linux, the runtime
cpu detection reads /proc/cpuinfo, so removing stdio.h from
celt/os_support.h meant that the cmake build for arm targets
failed.

We don't currently have ci runs for that configuration, so
this only became apparent through manual testing.

Signed-off-by: Marcus Asteborg <maastebo@microsoft.com>
Signed-off-by: Jean-Marc Valin <jmvalin@jmvalin.ca>
---
 celt/arm/armcpu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/celt/arm/armcpu.c b/celt/arm/armcpu.c
index 694a63b78..cce3ae3a9 100644
--- a/celt/arm/armcpu.c
+++ b/celt/arm/armcpu.c
@@ -93,6 +93,8 @@ static OPUS_INLINE opus_uint32 opus_cpu_capabilities(void){
 
 #elif defined(__linux__)
 /* Linux based */
+#include <stdio.h>
+
 opus_uint32 opus_cpu_capabilities(void)
 {
   opus_uint32 flags = 0;
